presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -914,6 +927,13 @@ private SetOperationPlan process(SetOperation node)
         return new SetOperationPlan(sources.build(), variableMapping.build());
     }
 
+    private void checkInterruption()
+    {
+        if (Thread.currentThread().isInterrupted()) {
+            throw new PrestoException(GENERIC_USER_ERROR, "Query plan timeout");
```

Comment:
I don't think this is the right error code.  It should have its own error code, and should probably be an internal or if we feel confident that it's a user issue, should be an insufficient resources error

Reply:
Done. Created a new Error code `QUERY_PLANNING_TIMEOUT` and output more detailed information "The query planner exceeded the timeout of xxx".